src/main/java/jewellery/inventory/service/ResourceInUserService.java

```diff
@@ -0,0 +1,116 @@
+package jewellery.inventory.service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import jewellery.inventory.dto.request.resource.ResourceInUserRequestDto;
+import jewellery.inventory.dto.response.resource.ResourceInUserResponseDto;
+import jewellery.inventory.exception.invalid_resource_quantity.InsufficientResourceQuantityException;
+import jewellery.inventory.exception.invalid_resource_quantity.NegativeResourceQuantityException;
+import jewellery.inventory.exception.not_found.ResourceInUserNotFoundException;
+import jewellery.inventory.exception.not_found.ResourceNotFoundException;
+import jewellery.inventory.exception.not_found.UserNotFoundException;
+import jewellery.inventory.mapper.UserMapper;
+import jewellery.inventory.model.User;
+import jewellery.inventory.model.resource.Resource;
+import jewellery.inventory.model.resource.ResourceInUser;
+import jewellery.inventory.repository.ResourceInUserRepository;
+import jewellery.inventory.repository.ResourceRepository;
+import jewellery.inventory.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ResourceInUserService {
+  private final UserRepository userRepository;
+  private final ResourceRepository resourceRepository;
+  private final ResourceInUserRepository resourceInUserRepository;
+
+  @Transactional
+  public ResourceInUserResponseDto addResourceToUser(ResourceInUserRequestDto resourceUserDto) {
+    User user = findUserById(resourceUserDto.getUserId());
+    Resource resource = findResourceById(resourceUserDto.getResourceId());
+
+    ResourceInUser resourceInUser =
+        findResourceInUser(user, resourceUserDto.getResourceId())
+            .orElseGet(() -> createAndAddNewResourceInUser(user, resource, 0));
+
+    resourceInUser.setQuantity(resourceInUser.getQuantity() + resourceUserDto.getQuantity());
+
+    return UserMapper.INSTANCE.toResourceInUserResponseDto(
+        resourceInUserRepository.save(resourceInUser));
+  }
+
+  public List<ResourceInUserResponseDto> getAllResourcesFromUser(UUID userId) {
```

Comment:
After the ResourcesInUserResponseDto, it would make this gets all the ResourceInUser owned by a user and returns a single ResourcesInUserResponseDto